src/datastore/Restaurant.cpp

```diff
@@ -0,0 +1,270 @@
+Restaurant::Restaurant(const string &Name, const float &Distance)
+        : m_Name(Name), m_fDistToSaddleback(Distance)
+{
+
+    m_fTotalPurchases = 0.0f;
+    m_nTotalCustomers = 0;
+    m_nUsedByCount = 0;
+    m_bDeleted = false;
+
+    m_nNumber =  sm_NextNumber++;
+
+    m_bInitialized = true;
+};
+
+// Constructor implementation
+Restaurant::Restaurant(
+         int        nNumber,
+         string     &Name,
+         float      &fDistToSaddleback,
+         float      &fTotalPurchases,
+         int        nTotalCustomers,
+         int        nUsedByCount,
+         bool       bDeleted,
+         vector<RestaurantDistance> &Distances,
+         vector<MenuItem>           &MenuItems,
+         vector<int>                &RefByTrips)
+: m_nNumber(nNumber), m_Name(Name), m_fDistToSaddleback(fDistToSaddleback),
+  m_fTotalPurchases(fTotalPurchases), m_nTotalCustomers(nTotalCustomers),
+  m_nUsedByCount(nUsedByCount), m_bDeleted(bDeleted),
+  m_Distances(Distances), m_Menu(MenuItems), m_ReferencedByTrips(RefByTrips)
+{
+    if (m_nNumber >= sm_NextNumber)
+    {
+        sm_NextNumber = m_nNumber + 1;
+    }
+    m_bInitialized = true;
+}
+
+// Destructor implementation
+Restaurant::~Restaurant()
+{
+}
+
+Restaurant::Restaurant(const Restaurant& src)
+: m_nNumber(src.m_nNumber), m_Name(src.m_Name), m_fDistToSaddleback(src.m_fDistToSaddleback),
+  m_fTotalPurchases(src.m_fTotalPurchases), m_nTotalCustomers(src.m_nTotalCustomers),
+  m_nUsedByCount(src.m_nUsedByCount), m_bDeleted(src.m_bDeleted),
+  m_Distances(src.m_Distances), m_Menu(src.m_Menu), m_ReferencedByTrips(src.m_ReferencedByTrips)
+{
+    m_bInitialized      = true;
+}
+
+Restaurant& Restaurant::operator=(const Restaurant& rhs)
+{
+   if (this == &rhs) {
+      return *this;
+   }
+
+    m_nNumber               = rhs.m_nNumber;
+    m_Name                  = rhs.m_Name;
+    m_fDistToSaddleback     = rhs.m_fDistToSaddleback;
+    m_fTotalPurchases       = rhs.m_fTotalPurchases;
+    m_nTotalCustomers       = rhs.m_nTotalCustomers;
+    m_nUsedByCount          = rhs.m_nUsedByCount;
+    m_bDeleted              = rhs.m_bDeleted;
+    m_Distances             = rhs.m_Distances;
+    m_Menu                  = rhs.m_Menu;
+    m_ReferencedByTrips     = rhs.m_ReferencedByTrips;
+
+    m_bInitialized      = true;
+
+   return *this;
+}
+
+const string &Restaurant::GetName(void) const
+{
+    return m_Name;
+}
+
+int Restaurant::GetNumber(void) const
+{
+    return m_nNumber;
+}
+
+const vector<MenuItem> &Restaurant::GetMenu(void) const
+{
+    return m_Menu;
+}
+
+const vector<RestaurantDistance> &Restaurant::GetDistances(void) const
+{
+    return m_Distances;
+}
+
+float Restaurant::GetPurchaseAmount(void) const
+{
+    return m_fTotalPurchases;
+}
+
+bool Restaurant::MarkDeleted(bool Delete)
+{
+    m_bDeleted = Delete;
+    return m_bDeleted;
+}
+
+bool Restaurant::AddTrip(int TripNumber)
+{
+    // should avoid adding duplicates
+    m_ReferencedByTrips.push_back(TripNumber);

```

Comment:
Function doesn't look finished. This solution should work as intended
```suggestion
    if(std::find(m_ReferencedByTrips.begin(), m_ReferencedByTrips.end(), TripNumber) == m_ReferencedByTrips.end())
    {
        m_ReferencedByTrips.push_back(TripNumber);
        return true;
    }
    
    return false;
```